src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs

```diff
@@ -22,7 +24,9 @@ private class Visitor : RazorIRNodeWalker
 
             public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
             {
-                if (string.Equals(node.Descriptor.TypeName, "Microsoft.AspNetCore.Mvc.ViewFeatures.ModelExpression", StringComparison.Ordinal))
+                if (string.Equals(node.Descriptor.TypeName, ModelExpressionTypeName, StringComparison.Ordinal) ||
+                    (node.IsIndexerNameMatch &&
+                     string.Equals(node.Descriptor.IndexerTypeName, ModelExpressionTypeName, StringComparison.Ordinal)))
```

Comment:
This was not a bug. We had specific descriptors for both the indexer and non-indexer case before the cleanup. But now we only have one descriptor for both. That is why this check is necessary.
We do have a test for this https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/RazorEngineTest.cs#L60